plaidml/bridge/openvino/ops/gather_tree.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+using edsl::Tensor;
+
+namespace PlaidMLPlugin {
+
+Tensor GatherTree(Tensor STEP_IDS, Tensor PARENT_IDX, Tensor MAX_SEQ_LEN, Tensor END_TOKEN) {
+  std::vector<int64_t> step_ids_shape = STEP_IDS.compute_shape().sizes();
+  std::vector<int64_t> parent_idx_shape = PARENT_IDX.compute_shape().sizes();
+  std::vector<int64_t> max_seq_len_shape = MAX_SEQ_LEN.compute_shape().sizes();
+  std::vector<int64_t> end_token_shape = END_TOKEN.compute_shape().sizes();
+
+  // Check shape
+  if (step_ids_shape.size() != 3) {
+    THROW_IE_EXCEPTION << "step_ids rank must be 3";
+  }
+  int max_time = step_ids_shape[0];
+  int batch_size = parent_idx_shape[1];
+  int beam_width = step_ids_shape[2];
+  if (max_seq_len_shape[0] != batch_size) {
+    THROW_IE_EXCEPTION << "max_seq_len and step_ids must have same batch size";
+  }
+  if (end_token_shape.size() != 0) {
+    THROW_IE_EXCEPTION << "end_token must be a scalar";
```

Comment:
Is it because scalar tensor has no shape so end_token_shape.size() == 0? What if it is a 0-D (one element) shaped tensor?

Reply:
If it only allows a scalar tensor here, maybe we need a `is_constant_tensor` function in shared header file. I think this is used in quite a few ops.

Reply:
This comes from https://github.com/plaidml/openvino/blob/plaidml/ngraph/core/src/op/gather_tree.cpp#L80. it is a tensor with zero shape but not must be a constant.  I think we can use 0-D tensor  like the ZERO in this code, but if we use openvino, may keep this